idaes/core/base/flowsheet_model.py

```diff
@@ -342,35 +367,18 @@ def _setup_dynamics(self):
                     "{} was set as a dynamic flowsheet, but time domain "
                     "provided was not a ContinuousSet.".format(self.name)
                 )
+
             add_object_reference(self, "_time", self.config.time)
             self._time_units = self.config.time_units
         else:
             # If no parent flowsheet, set up time domain
             if fs is None:
                 # Create time domain
-                if self.config.dynamic:
-                    # Check if time_set has at least two points
-                    if len(self.config.time_set) < 2:
-                        # Check if time_set is at default value
-                        if self.config.time_set == [0.0]:
-                            # If default, set default end point to be 1.0
-                            self.config.time_set = [0.0, 1.0]
-                        else:
-                            # Invalid user input, raise Exception
-                            raise DynamicError(
-                                "Flowsheet provided with invalid "
-                                "time_set attribute - must have at "
-                                "least two values (start and end)."
-                            )
-                    # For dynamics, need a ContinuousSet
-                    self._time = ContinuousSet(initialize=self.config.time_set)
-                else:
-                    # For steady-state, use an ordered Set
-                    self._time = pe.Set(initialize=self.config.time_set, ordered=True)
-                self._time_units = self.config.time_units
-
-                # Set time config argument as reference to time domain
-                self.config.time = self._time
+                _create_time_domain()
+            # Creates a new time domain for a sub-flowsheet with user provided time domain
+            elif self.config.time_set not in [None, [0]]:
```

Comment:
Why is the `[0]` here? I would have thought that if we are handed an argument we should use it. To put it another way, if the user provides `[0]` as an argument we will ignore it without saying why.